src/script/interpreter.cpp

```diff
@@ -1529,11 +1538,23 @@ class CTransactionSignatureSerializer
     /** Serialize an output of txTo */
     template<typename S>
     void SerializeOutput(S &s, unsigned int nOutput) const {
-        if (fHashSingle && nOutput != nIn)
+        if (fHashSingle && nOutput != nIn) {
             // Do not lock-in the txout payee at other indices as txin
             ::Serialize(s, CTxOut());
```

Comment:
Legacy HashSingle should commit to the range proofs since that is what the BIP143 variant does.

Reply:
It does right? Or is this something with the case where `nIn > txTo.vout.size()` ?

Reply:
Indeed it does.  I was misreading this clause.